janusgraph-test/src/test/java/org/janusgraph/graphdb/query/vertex/VertexIdListTest.java

```diff
@@ -0,0 +1,110 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.graphdb.query.vertex;
+
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+import org.janusgraph.core.JanusGraphVertex;
+import org.janusgraph.graphdb.internal.ElementLifeCycle;
+import org.janusgraph.graphdb.internal.InternalVertex;
+import org.janusgraph.graphdb.transaction.StandardJanusGraphTx;
+import org.janusgraph.graphdb.vertices.CacheVertex;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class VertexIdListTest {
+
+    @Mock
+    private StandardJanusGraphTx tx;
+
+    @Test
+    public void testMergeSortedLongIdLists() {
+        VertexIdList list = new VertexIdList(tx, Arrays.asList(1L, 2L, 3L), true);
+        VertexIdList otherList = new VertexIdList(tx, Arrays.asList(0L, 2L, 4L), true);
+        list.addAll(otherList);
+        assertEquals(Arrays.asList(0L, 1L, 2L, 2L, 3L, 4L), list.getIDs());
+        assertTrue(list.isSorted());
+    }
+
+    @Test
+    public void testMergeUnsortedLongIdLists() {
+        VertexIdList list = new VertexIdList(tx, new ArrayList<>(Arrays.asList(4L, 2L, 3L)), false);
+        VertexIdList otherList = new VertexIdList(tx, Arrays.asList(0L, 5L, 4L), false);
+        list.addAll(otherList);
+        assertEquals(Arrays.asList(4L, 2L, 3L, 0L, 5L, 4L), list.getIDs());
+        assertFalse(list.isSorted());
+    }
+
+    @Test
+    public void testMergeSortedStringIdLists() {
+        VertexIdList list = new VertexIdList(tx, Arrays.asList("x1", "x2", "y1"), true);
+        VertexIdList otherList = new VertexIdList(tx, Arrays.asList("x2", "x4"), true);
+        list.addAll(otherList);
+        assertEquals(Arrays.asList("x1", "x2", "x2", "x4", "y1"), list.getIDs());
+        assertTrue(list.isSorted());
+    }
+
+    @Test
+    public void testMergeSortedHeterogeneousIdLists() {
+        VertexIdList list = new VertexIdList(tx, Arrays.asList("x1", "x2", "y1"), true);
+        VertexIdList otherList = new VertexIdList(tx, Arrays.asList(2, 4), true);
+        list.addAll(otherList);
+        assertEquals(Arrays.asList(2, 4, "x1", "x2", "y1"), list.getIDs());
+        assertTrue(list.isSorted());
+    }
+
+    @Test
+    public void testMergeUnsortedHeterogeneousIdLists() {
+        VertexIdList list = new VertexIdList(tx, new ArrayList<>(Arrays.asList("x1", "x2", "y1")), true);
```

Comment:
Shouldn't this be:
```suggestion
        VertexIdList list = new VertexIdList(tx, new ArrayList<>(Arrays.asList("x1", "x2", "y1")), false);
```

Reply:
Nope. This is to showcase that merging a sorted list with an unsorted list would result in an unsorted list.

Reply:
Understood, the naming scheme just made me expect this to be basically the same as `testMergeUnsortedLongIdLists`, just with a mixture of strings and longs. You could maybe add this as an additional test specifically for the scenario of merging a sorted with an unsorted list to reduce this confusion, but it's also not really important.